utils/utils.go

```diff
@@ -37,12 +37,17 @@ const (
 	sishDNSPrefix = "sish="
 )
 
+type PublicKeysAndOptions struct {
+	options []string
+	key     ssh.PublicKey
+}
+
 var (
 	// Filter is the IPFilter used to block connections.
 	Filter *ipfilter.IPFilter
 
 	// certHolder is a slice of publickeys for auth.
-	certHolder = make([]ssh.PublicKey, 0)
+	certHolder = make([]PublicKeysAndOptions, 0)
```

Comment:
We can make this a map to make lookups easier instead of looping through it. Just have them key'd based on the ssh connection remote